stable/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -78,25 +77,29 @@ spec:
           protocol: TCP
         {{- end }}
         env:
+          - name: DD_HEALTH_PORT
+            value: {{ .Values.clusterAgent.healthPort | default 8080 | quote }}
           - name: DD_API_KEY
             valueFrom:
               secretKeyRef:
                 name: {{ template "datadog.apiSecretName" . }}
                 key: api-key
-          {{- if .Values.clusterAgent.metricsProvider.enabled }}
-          - name: DD_EXTERNAL_METRICS_PROVIDER_ENABLED
-            value: {{ .Values.clusterAgent.metricsProvider.enabled | quote }}
+                optional: true
           - name: DD_APP_KEY
             valueFrom:
               secretKeyRef:
                 name: {{ template "datadog.appKeySecretName" . }}
                 key: app-key
+                optional: true
```

Comment:
Should this be optional? Isn't it required for all communication back with DataDog

Reply:
it is requires if the user want the "external-metrics" feature, and the kubernetes events check.
Else it is not.

Reply:
Ah got it. Thanks for the clarification